### PR TITLE
Add boolean to change host value in createClient

### PR DESCRIPTION
**Pull request by bechatman**
This is a fix for #84 that worked for me. I wasn't able to get a host variable to pass through as an empty string, so I added `preview` boolean value to the request.

### CODE REVIEW

**Review by jescalan — COMMENTED** [review]
Fantastic! Love the proactive approach to the issue. Any chance of adding a test for this just to make sure there are no regressions in the future?

**Comment by bechatman** [comment]
@jescalan I added a few tests and fixed the ones that my change broke, but I haven't been able to get it to return valid content. I did add a new item to `.env` for previewToken, so that will need to be replaced by the preview token for that space.

**Comment by bechatman** [comment]
@jescalan Also, the downside of this change is that it would be a breaking change for existing implementations. There may be a better way to handle this that wouldn't be breaking, but I wasn't able to come up with anything.

**Comment by jescalan** [comment]
Hmm, so there's no reason I can see that this should be breaking. It's simply adding another option - if that option isn't used, the behavior shouldn't change. Is that not what you're experiencing?

**Comment by bechatman** [comment]
I went back to check, and you are right. I was using a ternary operator to switch between the two keys, so it was trying to use `host: ''` with the preview key, which doesn't work. I am going to push up a change to make `preview` optional, so that will fix the problem. The tests will keep failing until there is a preview key added, though.

**Comment by jescalan** [comment]
Do you have a preview key you are testing with locally? Or do we need to get one from contentful? The repo I'm using to test against is their public testing repo, so I'm sure they would be able to provide one if we reached out 😁

**Comment by bechatman** [comment]
I didn't rewrite any of the existing tests, so I didn't have a content model that would work. If you have access to the repo and get the preview key (or they can give it to you), that would be best.

**Comment by jescalan** [comment]
Hey @bechatman - sorry for the slight delay, I submitted a support ticket to contentful to see if they would share a preview key for their example repo that we use with these tests, will let you know when I hear back!

**Comment by jescalan** [comment]
Hey @bechatman! Got the preview key: `e5e8d4c5c122cf28fc1af3ff77d28bef78a3952957f15067bbc29f2f0dde0b50` - let me know if this works!

This is a public, read-only key so anyone else looking on, don't freak out about it being here 😁

**Comment by bechatman** [comment]
@jescalan Thanks! I will try it out now.

**Comment by bechatman** [comment]
@jescalan So the new tests passed with the preview key (thanks again for getting that), but a couple of the existing tests failed. I cloned the spike-contentful repo again and ran the tests on a clean copy, and those same tests failed. It looks like the "Implements request options" assumes an order that is no longer served by Contentful. Also, the default transform test just causes `locals` to be an empty string. 

I commented out the line that failed from the former (though changing the value it looks for to 'Nyan Cat' also fixes it), and I commented out the default transform test.

**Comment by jescalan** [comment]
Ok cool, I can fix those other tests!

File: test/fixtures/template/public/main.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/***/ 2:
+/***/ (function(module, exports, __webpack_require__) {
+
+(function webpackMissingModule() { throw new Error("Cannot find module \"/Users/brian/dev/lightpapers/spike-contentful/test/fixtures/template/main.js\""); }());
+
+
+/***/ })
+
+/******/ });
```
Comment (jescalan) [inline test/fixtures/template/public/main.js]: Can we remove these `public` folder compiled files?

File: test/index.js [hunk 2]
```diff
@@ -24,20 +24,21 @@ test('errors without a "spaceId"', (t) => {
...
+test('initializes with an "accessToken", "spaceId", and "addDataTo" and "preview"', (t) => {
+  const rt = new Contentful({ accessToken: 'xxx', spaceId: 'xxx', preview: false, addDataTo: {} })
   t.truthy(rt)
 })
 
 test('initializes with "limit" filter', (t) => {
   t.truthy(new Contentful({
     accessToken: 'xxx',
     spaceId: 'xxx',
+    preview: false,
```
Comment (jescalan) [inline test/index.js]: This should default to false

File: test/index.js [hunk 3]
```diff
@@ -59,7 +60,7 @@ test('errors with "limit" filter under 1', (t) => {
 test('errors with "limit" filter over 100', (t) => {
   t.throws(
     () => {
-      new Contentful({ accessToken: 'xxx', spaceId: 'xxx', addDataTo: {}, contentTypes: [{ // eslint-disable-line
+      new Contentful({ accessToken: 'xxx', spaceId: 'xxx', preview: false, addDataTo: {}, contentTypes: [{ // eslint-disable-line
```
Comment (jescalan) [inline test/index.js]: Same for these two - false should be the default unless it's set

File: test/index.js [hunk 4]
```diff
@@ -135,7 +165,7 @@ test.cb('implements request options', (t) => {
 
   api.run(undefined, () => {
     t.is(locals.contentful.cats.length, 1)
-    t.is(locals.contentful.cats[0].fields.name, 'Garfield')
+    /* t.is(locals.contentful.cats[0].fields.name, 'Garfield') /* change from Garfield to Nyan Cat passes test */
```
Comment (jescalan) [inline test/index.js]: Seems like they changed the fixture then, which is kind of an issue - you can switch it over!

**Review by jescalan — CHANGES_REQUESTED** [review]
Ok just a couple small modifications and we can get this guy merged!

File: test/index.js (hunk 3 above)
Comment (bechatman) [reply test/index.js]: @jescalan Does it need to be explicitly set? Because if it isn't present, it defaults to false as well.

Comment (bechatman) [reply test/index.js]: @jescalan I think that might be remnant of my misunderstanding how to set up the tests, I will try removing the reference entirely except in tests that invoke the preview.

**Comment by bechatman** [comment]
@jescalan I made the updates but the cover test still fails because of commenting out that `transform: true` test.

**Comment by bechatman** [comment]
@jescalan Any thoughts on how to update the tests to get the Coveralls check to pass? Looks like it is just that default transform test causing issues. Not sure if changes to Spike makes that test fail or what.

**Comment by jescalan** [comment]
I'm sorry, I'm not entirely sure. I'll pull this down and take a look now

**Comment by jescalan** [comment]
Ok just pushed some new code to master that updates deps, removes the transform function, and all tests are passing. You can rebase on top of that and we should be good to go!

**Comment by bechatman** [comment]
@jescalan I got everything merged in, but the preview test failed. I tried the updated version of spike contentful with the preview function added on my sites and they worked, so I commented out the other tests and updated them to work against my content and with my keys. The tests passed when I did that. Is it possible the preview key was only valid for a specific time period?

**Comment by jescalan** [comment]
@bechatman I highly doubt it - the same key is used as part of contentful's own tests. If so, then their tests would be failing.

https://github.com/contentful/contentful.php/blob/6a28cabbd42c9e6995aa50d576bdf58fa1093694/tests/End2EndTestCase.php#L43

**Comment by bechatman** [comment]
@jescalan that looks like a longee key. Let me see if I just have the wrong key in there (though I don’t know why it would have worked before)

**Comment by bechatman** [comment]
@jescalan Yeah I think the preview key got messed up somewhere along the way. That test is passing. I messed up a merge conflict resolution in the test file, too, so I just copied yours and added the preview test back in.

**Comment by bechatman** [comment]
@jescalan I am going to make a readme modification so this is documented now that the tests are passing.

**Review by jescalan — APPROVED** [review]
Looking great! Will get this merged up next week 🙌 

Thanks for your continued hard work on this feature 🙇

**Comment by bechatman** [comment]
@jescalan Thanks! This is my first contribution to an open-source project, so thanks for being patient.

**Comment by jescalan** [comment]
Released as `v3.1.0` - thanks again for the hard work on this contribution! 🙌 🎊 👬

**Comment by bechatman** [comment]
Awesome! Thank you!
